lib/image-stream/parse-partitions.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const Bluebird = require('bluebird');
+const MBR = require('mbr');
+const GPT = require('gpt');
+
+const MAX_BYTES = 65536;
+const BLOCK_SIZE = 512;
+const ZERO = 0;
+
+/**
+ * Get the GPT/EFI partition from a MBR
+ * @param {MBR} mbr - Master Boot Record
+ * @returns {MBR.Partition|undefined} partition
+ */
+const getEfiPart = (mbr) => {
+  const GPT_HYBRID = 0xEE;
+  const GPT_PROTECTIVE = 0xEF;
+  return _.filter(mbr.partitions, (partition) => {
```

Comment:
This sounds like it should be a `_.find`, which allows you to get rid of the `[ZERO]`